addons/lunch/models/lunch_supplier.py

```diff
@@ -195,6 +195,8 @@ def write(self, values):
         if values.get('company_id'):
             self.env['lunch.order'].search([('supplier_id', 'in', self.ids)]).write({'company_id': values['company_id']})
         super().write(values)
+        # flush fields to call _sql_constraints
+        self.flush()
```

Comment:
I guess it should be moved inside if-block.

Also, what about create method? It also uses `_sync_cron`

Reply:
Are you saying that `_sql_constraints` is used in create method without `flush()`?